frontend/src/pages/RecipientRecord/pages/CommunicationLog.js

```diff
@@ -1,8 +1,97 @@
-import React from 'react';
+import React, { useState, useContext } from 'react';
 import PropTypes from 'prop-types';
 import { Helmet } from 'react-helmet';
 
-export default function CommunicationLog({ recipientName, regionId }) {
+import { Link } from 'react-router-dom';
+import useDeepCompareEffect from 'use-deep-compare-effect';
+import { getCommunicationLogsByRecipientId } from '../../../fetchers/communicationLog';
+import AppLoadingContext from '../../../AppLoadingContext';
+import WidgetContainer from '../../../components/WidgetContainer';
+import CommunicationLogTable from './components/CommunicationLogTable';
+
+const COMMUNICATION_LOG_PER_PAGE = 10;
+
+export default function CommunicationLog({ recipientName, regionId, recipientId }) {
+  const [logs, setLogs] = useState();
+  const [error, setError] = useState();
+  const [sortConfig, setSortConfig] = useState({
+    sortBy: 'communicationDate',
+    direction: 'desc',
+    offset: 0,
+    activePage: 1,
+  });
+
+  const { setIsAppLoading } = useContext(AppLoadingContext);
+
+  useDeepCompareEffect(() => {
```

Comment:
Why do we need deep compare vs regular use effect?

Reply:
My guess is because of the `sortConfig`. We want to re-fetch when something inside this object changes (deep compare) instead of just when the object changes (reference compare, like normal useEffect)

Reply:
jon is correct, we want to deep compare sortConfig

Reply:
NVM makes sense.